com.civicrm.hrjobroles/js/src/job-roles/services/date-validation.js

```diff
@@ -62,25 +92,34 @@ define([
              * @param {Date|string|int} start
              * @param {Date|string|int} end
              */
-            validate: function validate(start, end) {
-                if (start instanceof Date) {
-                    start = start.getTime();
-                }
-
-                if (end instanceof Date) {
-                    end = end.getTime();
-                }
+            validate: function validate(start, end, contract_start, contract_end) {
+                start = formatDate(start);
```

Comment:
What's the need for having both `start` and `start_date` (same goes for `end`)? Couldn't you have just one of the two (basically what you do for `contract_start` and `contract_end`?

If you could, you could simplify the code:
```js
validate: function validate(start, end, contract_start, contract_end) {
    start = moment(formatDate(start), this.dateFormats, true);
    end = moment(formatDate(end), this.dateFormats, true);
    contract_start = moment(formatDate(contract_start), this.dateFormats, true);
    contract_end = moment(formatDate(contract_end), this.dateFormats, true);

    checkIfValuesAreValid(start, ['start_date']);
    // ....
}

(Although I noticed that the original code sometimes checks `start`, sometimes `start_date`)
```